projects/git-scm.org/git-shim

```diff
@@ -2,26 +2,43 @@
 
 set -e
 
-find_git_command() {
+# This function searches through the provided arguments
+# to find the first non-option argument, which is assumed
+# to be the git subcommand.
+find_git_subcommand() {
+  skip_next=0  # Flag to indicate if the next argument should be skipped.
+
   for arg in "$@"; do
-    if test "$SKIP" = 1; then
-      SKIP=0
+    # If the skip_next flag is set, skip this argument.
+    if [ "$skip_next" -eq 1 ]; then
+      skip_next=0
       continue
     fi
+
+    # Check for options that require skipping the next argument.
     case $arg in
-    -C) SKIP=1; continue;;
-    -*);;
-    *)
-      echo "$arg"
-      return  # found the command!
+      -C | --git-dir | --work-tree | --namespace | --super-prefix | --config-env | -c)
+        skip_next=1  # The next argument is a value for these options.
+        continue
+        ;;
+      --exec-path=* | -c=* | --git-dir=* | --work-tree=* | --namespace=* | --super-prefix=* | --config-env=*)
+        continue  # Skip options that include their values with '='.
+        ;;
+      -*)
+        continue  # Skip other options that start with '-'.
```

Comment:
this first case will be handled by the second (i.e. doesn't need an additional case).

Reply:
Ah, yes. Fixed!